sstables/compaction.cc

```diff
@@ -1272,22 +1297,13 @@ GCC6_CONCEPT(
 )
 future<compaction_info> compaction::run(std::unique_ptr<compaction> c, GCConsumer gc_consumer) {
     return seastar::async([c = std::move(c), gc_consumer = std::move(gc_consumer)] () mutable {
-        auto reader = c->setup();
-
-        auto cr = c->get_compacting_sstable_writer();
-        auto cfc = make_stable_flattened_mutations_consumer<compact_for_compaction<compacting_sstable_writer, GCConsumer>>(
-                    *c->schema(), gc_clock::now(), c->max_purgeable_func(), std::move(cr), std::move(gc_consumer));
-
+        auto consumer = c->setup(std::move(gc_consumer));
```

Comment:
Could have passed both consumers, instead of moving one into the method and leaving the other one outside.

Reply:
could be lack of github context but I don't even know which two consumers you are talking about

Reply:
The code removed around this snippet and moved inside the `setup()` call.

Reply:
Still don't know what you mean.
There is only one consumer pre-existing the call to setup, the `gc_consumer`.

The other one is returned from setup, how could I have passed it?

Reply:
Its soooooooooooo painful to locate comments that have updates.

You are right. Setup wants to create the consumer itself.